Assets/Tests/InputSystem/Plugins/UITests.cs

```diff
@@ -223,6 +245,194 @@ public IEnumerator MouseActions_CanDriveUI()
+        InputSystem.Update();
+
+        foreach (var player in players)
+        {
+            Assert.That(player.leftChildReceiver.events, Has.Count.EqualTo(0));
+            Assert.That(player.rightChildReceiver.events, Has.Count.EqualTo(0));
+            player.eventSystem.InvokeUpdate();
+        }
+
+        Assert.That(players[0].eventSystem.currentSelectedGameObject, Is.EqualTo(players[0].rightGameObject));
+        Assert.That(players[1].eventSystem.currentSelectedGameObject, Is.EqualTo(players[1].leftGameObject));
+
+        Assert.That(players[0].leftChildReceiver.events, Has.Count.EqualTo(2));
+        Assert.That(players[0].leftChildReceiver.events[0].type, Is.EqualTo(EventType.Move));
+        Assert.That(players[0].leftChildReceiver.events[1].type, Is.EqualTo(EventType.Deselect));
+        players[0].leftChildReceiver.Reset();
+
+        Assert.That(players[0].rightChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(players[0].rightChildReceiver.events[0].type, Is.EqualTo(EventType.Select));
+        players[0].rightChildReceiver.Reset();
+
+        foreach (var player in players)
+        {
+            Assert.That(player.leftChildReceiver.events, Has.Count.EqualTo(0));
+            Assert.That(player.rightChildReceiver.events, Has.Count.EqualTo(0));
+            player.eventSystem.InvokeUpdate();
+        }
+
+        // Check Player 0 Submit
+        InputSystem.QueueStateEvent(gamepads[0], new GamepadState { buttons = 1 << (int)GamepadButton.South });
+        InputSystem.Update();
+
+        foreach (var player in players)
+        {
+            Assert.That(player.leftChildReceiver.events, Has.Count.EqualTo(0));
+            Assert.That(player.rightChildReceiver.events, Has.Count.EqualTo(0));
+            player.eventSystem.InvokeUpdate();
+        }
+
+        Assert.That(players[0].rightChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(players[0].rightChildReceiver.events[0].type, Is.EqualTo(EventType.Submit));
+        players[0].rightChildReceiver.Reset();
+
+        // Check Player 1 Submit
+        InputSystem.QueueStateEvent(gamepads[1], new GamepadState { buttons = 1 << (int)GamepadButton.South });
+        InputSystem.Update();
+
+        foreach (var player in players)
+        {
+            Assert.That(player.leftChildReceiver.events, Has.Count.EqualTo(0));
+            Assert.That(player.rightChildReceiver.events, Has.Count.EqualTo(0));
+            player.eventSystem.InvokeUpdate();
+        }
+        Assert.That(players[1].leftChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(players[1].leftChildReceiver.events[0].type, Is.EqualTo(EventType.Submit));
+        players[1].leftChildReceiver.Reset();
+
+        foreach (var player in players)
+        {
+            Assert.That(player.leftChildReceiver.events, Has.Count.EqualTo(0));
+            Assert.That(player.rightChildReceiver.events, Has.Count.EqualTo(0));
+        }
+    }
```

Comment:
Don't remember what the test framework does with the scene. Will it clean up the scene? I vaguely remember it resets the scene only once and then whatever each test spills into the scene will stick around. Wondering if we can get side-effects between the tests here?

Reply:
Yes, it seems to clean this  up between tests.